prefq/examples/imitation_preference_comparisons.py

```diff
@@ -74,16 +75,35 @@ def gather(self) -> Tuple[Sequence[TrajectoryWithRewPair], np.ndarray]:
                 output_path=os.path.join(self.video_dir, f"{query_id}-right.webm"),
             )
 
+            empty_query_response = {query_id: None}
+            query_dict.update(empty_query_response)
             self._send_videos_to_server(query_id)
 
         
         feedback_data = self._get_feedback_from_server()
 
+        print("\n\nUnordered Query Response")
         for query_id, is_left_preferred in feedback_data.items():
             print(f"    Query ID: {query_id}    Left Video Preferred: {is_left_preferred}")
-            preferences = np.zeros(len(self.pending_queries), dtype=np.float32)
+
+            # Order of queries and preferences must match, but the server response is unordered
+            # due to:
+            #   (1) The asynchronous nature feedback collection (multiple users can send feedback)
+            #   (2) The nature of the flask.jsonify() function, that orders keys alphanumerically.
+            #       This behavior can be turned off, but makes no sense due to (1)
+            query_dict[query_id] = is_left_preferred
+
+        print("\n\nOrdered Query Response")
+        for query_id, is_left_preferred in query_dict.items():
```

Comment:
We should not rely on the order of a dictionary. Although dictionaries technically have an order, it is not really intended to be an ordered datatype and in my opinion it is not the best style to rely on it.

I propose to just keep two dictionaries, one for queries and one for labels. Both are indexed by query IDs. In the end, we can iterate over all query IDs and create two list (`queries` and `preferences`). Something like

```python
queries = []
preferences = []
for query_id in feedback_data.keys():
    queries.append(pending_queries[query_id])
    preferences.append(feedback_data[query_id])
```